Add constraints to auth app config

scotttrinh opened the pull request:
Since these values end up in user-facing templates, ensure that they are limited to sane lengths and are also well sanitized.

1st1 commented on edb/server/protocol/auth_ext/util.py:
```diff
@@ -101,28 +103,38 @@ def get_config_typename(config_value: edb_config.SettingValue) -> str:
+def escape_and_truncate(input_str: str | None, max_len: int) -> str | None:
+    if input_str is None:
+        return None
+    return html.escape(input_str)[:max_len]
```
escape should happen on the trimmed string, not after -- can risk having broken escape codes

1st1 reviewed (commented):
Can you add tests by any chance? Can be a unit test

1st1 commented on edb/server/protocol/auth_ext/util.py:
```diff
@@ -101,28 +103,38 @@ def get_config_typename(config_value: edb_config.SettingValue) -> str:
+def escape_and_truncate(input_str: str | None, max_len: int) -> str | None:
+    if input_str is None:
+        return None
+    return html.escape(input_str[:max_len])
```
```suggestion
    trunc = (
        f"{input_str[:max_len]}..." 
        if len(input_str) > max_len else
        input_str
    )
    return html.escape(trunc)
```

scotttrinh commented:
@1st1 ⚡ 

Added tests for both the UI and the WebAuthn relying party info (we weren't truncating and escaping there), and added the suggested change showing ellipsis.
